src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Patch.txt

```diff
@@ -0,0 +1,56 @@
+﻿#SETUP: EnableExpressionChaining,MutationFunctionsTestSetup
+
+// MutationFunctionsTestSetup handle does the following:
+// - Creates a record: r1 => {Field1:1,Field2:"earth",Field3:1/1/2022,Field4:true}
+// - Creates a record: r2 => {Field1:2,Field2:"moon",Field3:2/1/2022,Field4:false} (for convenience)
+// - Creates a record: r_empty => {}
+// - Creates a table: t1(r1) => Type (Field1, Field2, Field3, Field4)
+
+>> Patch(r1, {Field5:5})
+{Field1:1,Field2:"earth",Field3:DateTime(2022,1,1,0,0,0,0),Field4:true,Field5:5}
+
+>> Patch(r1, r_empty)
+{Field1:1,Field2:"earth",Field3:DateTime(2022,1,1,0,0,0,0),Field4:true}
+
+>> Patch(r1, {Field2:"mars"})
+{Field1:1,Field2:"mars",Field3:DateTime(2022,1,1,0,0,0,0),Field4:true}
+
+>> Patch(r1, {Field2:"mars"}, Blank())
+{Field1:1,Field2:"mars",Field3:DateTime(2022,1,1,0,0,0,0),Field4:true}
+
+>> Patch(r1, {Field3:DateTime(2022,12,12,0,0,0,0)}, {Field4:false})
+{Field1:1,Field2:"earth",Field3:DateTime(2022,12,12,0,0,0,0),Field4:false}
+
+>> Patch(r1, r2)
+{Field1:2,Field2:"moon",Field3:DateTime(2022,2,1,0,0,0,0),Field4:false}
+
+>> Patch(t1, r1, r_empty);t1
+Table({Field1:1,Field2:"earth",Field3:DateTime(2022,1,1,0,0,0,0),Field4:true})
+
+>> Patch(t1, r1, {Field2:"mars"});t1
+Table({Field1:1,Field2:"mars",Field3:DateTime(2022,1,1,0,0,0,0),Field4:true})
+
+>> Patch(t1, r1, {Field2:"mars"}, Blank());t1
+Table({Field1:1,Field2:"mars",Field3:DateTime(2022,1,1,0,0,0,0),Field4:true})
+
+>> Patch(t1, r1, {Field3:DateTime(2022,12,12,0,0,0,0)}, {Field4:false});t1
+Table({Field1:1,Field2:"earth",Field3:DateTime(2022,12,12,0,0,0,0),Field4:false})
+
+>> Patch(t1, r1, r2);t1
+Table({Field1:2,Field2:"moon",Field3:DateTime(2022,2,1,0,0,0,0),Field4:false})
+
+>> Patch(t1, r1, {Field5:"Field5"}); t1
+Errors: Error 0-32: The function 'Patch' has some invalid arguments.|Error 14-31: The specified column 'Field5' does not exist. The column with the most similar name is 'Field1'.
+
+// This test would fail in PA.
```

Comment:
>This test would fail in PA. [](http://example.com/codeflow?start=3&length=27)

Why?...
Add some explanation

Reply:
"The data source supplied to the function is invalid"
I'll add the explanation in a future interaction.